src/native/fs/fs_napi.cpp

```diff
@@ -284,66 +432,82 @@ struct FSWrapWorker : public FSWorker
 struct Stat : public FSWorker
 {
     std::string _path;
+    bool _use_lstat = false;
+    bool _skip_user_xattr = false;
     struct stat _stat_res;
+    XattrMap _xattr;
 
     Stat(const Napi::CallbackInfo& info)
         : FSWorker(info)
     {
         _path = info[1].As<Napi::String>();
+        if ((int)info.Length() == 3) {
+            Napi::Object options = info[2].As<Napi::Object>();
+            if (options.Has("use_lstat")) _use_lstat = options.Get("use_lstat").ToBoolean();
+            if (options.Has("skip_user_xattr")) _skip_user_xattr = options.Get("skip_user_xattr").ToBoolean();
+        }
         Begin(XSTR() << "Stat " << DVAL(_path));
     }
     virtual void Work()
     {
-        int r = stat(_path.c_str(), &_stat_res);
+        int flags = O_RDONLY;
+        // using O_PATH (MAC - O_SYMLINK) with O_NOFOLLOW allow us to open the symlink itself
+        // instead of openning the file it links to https://man7.org/linux/man-pages/man7/symlink.7.html
+#ifdef __APPLE__
+        if (_use_lstat) flags = O_SYMLINK | O_NOFOLLOW;
+#else
+        if (_use_lstat) flags = O_PATH | O_NOFOLLOW;
+#endif
+        int fd = open(_path.c_str(), flags);
+        if (fd < 0) {
+            SetSyscallError();
+            return;
+        }
+        int r;
+        r = fstat(fd, &_stat_res);
+        if (r) {
+            SetSyscallError();
+            r = close(fd);
+            if (r) SetSyscallError(); // Report only
+            return;
+        }
+        if (!_use_lstat) { 
```

Comment:
can you add a comment to explain why use_lstat does not require (or does not support?) xattrs